source/Generic/DateImporter/DateImporter.psm1

```diff
@@ -846,5 +883,123 @@ function Get-SteamLicenses
         }
         $LicensesList.Add($License)
     }
+    return $LicensesList
+}
+
+function Export-OriginLicenses
+{
+    param(
+        $scriptMainMenuItemActionArgs
+    )
+    
+    $libraryName = "Origin"
+    
+    $LicensesList = Get-OriginLicenses $libraryName
+    if ($LicensesList.count -eq 0)
+    {
+        $PlayniteApi.Dialogs.ShowMessage([Playnite.SDK.ResourceProvider]::GetString("LOCDate_Importer_LicensesNotFoundMessage"), "$libraryName Date Importer")
+        return
+    }
+    Export-Licenses $libraryName $LicensesList
+}
+
+function Add-OriginDates
+{
+    param (
+        $gameDatabase
+    )
+
+    if ($gameDatabase.count -eq 0)
+    {
+        return
+    }
+
+    $libraryName = "Origin"
+    $LicensesList = Get-OriginLicenses $libraryName
+    if ($LicensesList.count -eq 0)
+    {
+        $PlayniteApi.Dialogs.ShowMessage([Playnite.SDK.ResourceProvider]::GetString("LOCDate_Importer_LicensesNotFoundMessage"), "$libraryName Date Importer")
+        return
+    }
+
+    Set-DatesFromLicenses $libraryName $LicensesList $gameDatabase
+}
+
+function Get-OriginLicenses
+{
+    param (
+        $libraryName
+    )
+    
+    [System.Collections.Generic.List[object]]$LicensesList = @()
+
+    $authUrl = "https://accounts.ea.com/connect/auth?client_id=ORIGIN_JS_SDK&response_type=token&redirect_uri=nucleus:rest&prompt=none"
+    $loginStatus = Get-LoginStatusViaJson $authUrl
+    if ($loginStatus -eq $false)
+    {
+        $PlayniteApi.Dialogs.ShowMessage(([Playnite.SDK.ResourceProvider]::GetString("LOCDate_Importer_UsetNotLoggedMessage") -f $libraryName), "$libraryName Date Importer")
+        return $LicensesList
+    }
+
+    $webView = $PlayniteApi.WebViews.CreateOffscreenView()
+    $webView.NavigateAndWait($authUrl)
+    $pageSource = $webView.GetPageSource()
+    $webView.Dispose()
+
+    $jsonValid = Get-IsJsonValidFromPage $pageSource
+
+    if($jsonValid -eq $false) 
+    {
+        $PlayniteApi.Dialogs.ShowMessage([Playnite.SDK.ResourceProvider]::GetString("LOCDate_Importer_MenuItemImportOriginInvalidJson"), "$libraryName Date Importer")
+        return $LicensesList
+    }
+
+    $authJson = Get-JsonFromPageSource $pageSource
+    $token = $authJson.access_token
+
+
+    $idUrl = "https://gateway.ea.com/proxy/identity/pids/me"
+    $headers = @{
+        Authorization = "Bearer {0}" -f $token
+    }
+    $idResponse = $null
+    try {
+        $idResponse = Invoke-RestMethod -Method "GET" -Uri $idUrl -Headers $headers -ContentType "application/json"
```

Comment:
Can you please change the request to use a net class instead of this? For the reason that it will fail if internet explorer is not configured, see: https://stackoverflow.com/questions/38005341/the-response-content-cannot-be-parsed-because-the-internet-explorer-engine-is-no

Or maybe check if it works with `-UseBasicParsing` as mentioned there

Reply:
Done in 85f6614eb5b9b7fa4801790c894fb5a2fd724f1f.
I added the -UseBasicParsing and it still works. Seems it's no longer neccessary for Powershell 6 and higher (https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-restmethod?view=powershell-7.1) but good call either way. It would be an easy enough way to support older installations. 
Can't really test it as funny enough I'm not even on Powershell 6 and i don't encounter the problem. I also have IE uninstalled and it still worked (with and without -UseBasicParsing) 😄

Reply:
I could reproduce it before I configured IE and had reports of several people having errors for that reason before so I've switched to using net class since then. I just tested it and there are no problems. If someone reports an issue then we can change it.